MiniApp/Classes/API/ListingAPI.swift

```diff
@@ -5,17 +5,26 @@ internal class ListingApi {
         self.environment = environment
     }
 
-    func createURLRequest() -> URLRequest? {
-        guard let url = getListingURL() else {
+    func createURLRequest(miniAppID: String? = nil) -> URLRequest? {
+        guard let url = getListingURL(miniAppID) else {
             return nil
         }
         return URLRequest.createURLRequest(url: url, environment: environment)
     }
 
-    private func getListingURL() -> URL? {
+    private func getListingURL(_ miniAppId: String? = nil) -> URL? {
```

Comment:
Can this be broken into smaller units, I see multiple `returns`?

Reply:
I did this to keep compatibility with existing code. The new parameter miniAppId is optional and nil if not provided